### ISSUE TITLE
[analyzer] No error if an extension type declares getter and setter with the same basename but with uncompatible types

### ISSUE THREAD

**Issue by sgrekhov** [dart-lang/sdk#53489]
Analyzer doesn't report an error if an extension type declares an instance getter and setter with incompatible types.
For example the following errors are not reported.
```dart
extension type ET1(int id) {
  void set x(String i) {}
  int get x => 42;
//        ^
// [analyzer] unspecified
// [cfe] unspecified   
}

extension type ET2(int id) {
  void set id(String i) {} // conflict with the type of representation variable getter
//         ^^
// [analyzer] unspecified
// [cfe] unspecified
}
```
cc @eernstg  to confirm that extension type should behave like a regular class here
Tested on the edge SDK (Sept 11, 2023) on Linux x64

**Comment by eernstg** [comment]
Yes, that error should be reported.

Cf. https://github.com/dart-lang/language/pull/3338.

**Comment by scheglov** [comment]
This was fixed in 08c010868567ae9751e0c59b61a2224af12be4d3.
